includes/core/classes/class-user.php

```diff
@@ -72,6 +72,10 @@ protected function setup_hooks(): void {
 	public function profile_fields( WP_User $user ): void {
 		$event_updates_opt_in = get_user_meta( $user->ID, 'gp-event-updates-opt-in', true );
 
+		$default_tz   = static::get_system_timezone();
+		$date_default = get_option( 'date_format', 'l, F j, Y' );
+		$time_default = get_option( 'time_format', 'g:i A' );
```

Comment:
GatherPress has it's own settings API you can use to get this data from what is set in those settings.
```
$settings     = Settings::get_instance();
$data_default = $settings->get_value( 'general', 'formatting', 'date_format' );
$time_default = $settings->get_value( 'general', 'formatting', 'time_format' );
```